tests/interfaces/records/handlers/records-delete.spec.ts

```diff
@@ -120,6 +121,376 @@ describe('RecordsDeleteHandler.handle()', () => {
+        requester : alice,
+        filter    : { recordId: aliceWriteData.message.recordId }
+      });
+      const aliceQueryWriteAfterBobWriteReply = await dwn.processMessage(alice.did, aliceQueryWriteAfterBobWriteData.message);
+      expect(aliceQueryWriteAfterBobWriteReply.status.code).to.equal(200);
+      expect(aliceQueryWriteAfterBobWriteReply.entries.length).to.equal(1);
+      expect(aliceQueryWriteAfterBobWriteReply.entries[0].encodedData).to.equal(encodedData);
+
+      const bobQueryWriteAfterBobWriteData = await TestDataGenerator.generateRecordsQuery({
+        requester : bob,
+        filter    : { recordId: bobWriteData.message.recordId }
+      });
+      const bobQueryWriteAfterBobWriteReply = await dwn.processMessage(bob.did, bobQueryWriteAfterBobWriteData.message);
+      expect(bobQueryWriteAfterBobWriteReply.status.code).to.equal(200);
+      expect(bobQueryWriteAfterBobWriteReply.entries.length).to.equal(1);
+      expect(bobQueryWriteAfterBobWriteReply.entries[0].encodedData).to.equal(encodedData);
+
+      await expect(fromAsync(dataPartition.db.keys())).to.eventually.eql([ dataCid ]);
+    });
+
+    it('should not allow referencing data across tenants', async () => {
+      const dataPartition = await dataStore.blockstore.partition('data');
+
+      const data = Encoder.stringToBytes('test');
+      const dataCid = await Cid.computeDagPbCidFromBytes(data);
+      const encodedData = Encoder.bytesToBase64Url(data);
+
+      const alice = await DidKeyResolver.generate();
+      const bob = await DidKeyResolver.generate();
+
+      const aliceWriteData = await TestDataGenerator.generateRecordsWrite({
+        requester: alice,
+        data
+      });
+      const aliceWriteReply = await dwn.processMessage(alice.did, aliceWriteData.message, aliceWriteData.dataStream);
+      expect(aliceWriteReply.status.code).to.equal(202);
+
+      const aliceQueryWriteAfterAliceWriteData = await TestDataGenerator.generateRecordsQuery({
+        requester : alice,
+        filter    : { recordId: aliceWriteData.message.recordId }
+      });
+      const aliceQueryWriteAfterAliceWriteReply = await dwn.processMessage(alice.did, aliceQueryWriteAfterAliceWriteData.message);
+      expect(aliceQueryWriteAfterAliceWriteReply.status.code).to.equal(200);
+      expect(aliceQueryWriteAfterAliceWriteReply.entries.length).to.equal(1);
+      expect(aliceQueryWriteAfterAliceWriteReply.entries[0].encodedData).to.equal(encodedData);
+
+      await expect(fromAsync(dataPartition.db.keys())).to.eventually.eql([ dataCid ]);
+
+      const bobAssociateData = await TestDataGenerator.generateRecordsWrite({
+        requester : bob,
+        dataCid,
+        dataSize  : 4
+      });
+      const bobAssociateReply = await dwn.processMessage(bob.did, bobAssociateData.message, bobAssociateData.dataStream);
+      expect(bobAssociateReply.status.code).to.equal(400);
```

Comment:
Just to make sure it's not some other error throwing 400 due to buggy test whatnot (happens more than I'd admit when I write tests)

```suggestion
      expect(bobAssociateReply.status.code).to.equal(400);
      expect(bobAssociateReply.status.detail).to.contain(DwnErrorCode.MessageStoreDataNotFound);
```